# Atividade 2

- Aluno: Leonardo H de Braz
- Matrícula: 217148

## Arquivos - Datasets

Aqui estão definidos os links para acesso dos datasets fornecidos no exercício

In [57]:
dataset_np = "https://raw.githubusercontent.com/lhleonardo/MO911-redacao-cientifica/main/bp.csv"
dataset_pr = "https://raw.githubusercontent.com/lhleonardo/MO911-redacao-cientifica/main/paired.csv"

## Configuração do ambiente

In [58]:
import matplotlib.pyplot as plt
import scipy.stats as stats
import pandas as pd
import numpy as np
import csv

In [59]:
dados_pareado     = pd.read_csv(dataset_pr)
dados_nao_pareado = pd.read_csv(dataset_np)

In [60]:
dados_pareado

Clone  August  November
0     Balsam_Spire     8.1      11.2
1          Beaupre    10.0      16.3
2        Hazendans    16.5      15.3
3        Hoogvorst    13.6      15.6
4         Raspalje     9.5      10.5
5             Unal     8.3      15.5
6   Columbia_River    18.3      12.7
7    Fritzi_Pauley    13.3      11.1
8        Trichobel     7.9      19.9
9            Gaver     8.1      20.4
10          Gibecq     8.9      14.2
11           Primo    12.6      12.7
12       Wolterson    13.4      36.8

In [61]:
dados_nao_pareado

bp type
0    68   No
1    70  Yes
2    82   No
3    76   No
4    60   No
..   ..  ...
195  58   No
196  68  Yes
197  70   No
198  58   No
199  62  Yes

[200 rows x 2 columns]

## Questão 1

O arquivo bp.csv contem dados sobre a pressão sanguínea (bp) e se a pessoa é ou não diabética (type) (diabetico é type = Yes).

Use os seguintes testes e responda os itens abaixo:

- Teste t

- Wilcoxon rank sum

1. Reporte o p-valor da comparação entre a pressão sangüínea de pessoas diabéticas ou não.

2. Qual tem a maior pressão média? E maior pressão mediana?

3. A diferença é significativa?

Discuta a diferença dos p-valores. Quais dos 2 testes você acha mais apropriado nesse caso?

### Teste T para análise de pressão sanguínea

In [62]:
diabeticos = dados_nao_pareado[dados_nao_pareado.type == 'Yes']
nao_diabeticos = dados_nao_pareado[dados_nao_pareado.type == 'No']

In [63]:
print(len(diabeticos))
print(len(nao_diabeticos))

68
132


In [64]:
stats.ttest_ind(diabeticos.bp, nao_diabeticos.bp)

Ttest_indResult(statistic=3.001495202122035, pvalue=0.0030323591864622732)

In [65]:
print(np.mean(diabeticos.bp))
print(np.mean(nao_diabeticos.bp))

74.58823529411765
69.54545454545455


In [66]:
print(np.median(diabeticos.bp))
print(np.median(nao_diabeticos.bp))

76.0
70.0


In [67]:
stats.ranksums(diabeticos.bp, nao_diabeticos.bp)

RanksumsResult(statistic=3.045797809294312, pvalue=0.0023206383267846886)

O p-value para o T-test foi 0.0030323591864622732.

O p-value para o Wilcoxon rank sum foi 0.0023206383267846886.

A maior pressão média foi encontrada para os diabéticos, com o valor de 74.58823529411765.

A maior pressão mediana foi encontrada para os diabéticos, com o valor de 76.0.

A diferença dos dois testes foi baixa, mas não é preciso utilizar o Wilcoxon neste caso. Como as medições foram aplicadas em pessoas diferentes (68 diabéticos e 132 não diabéticos), não existem valores "pareados", tornando o teste T-Student suficiente para provar que as populações são distintas.

## Tarefa 2

O arquivo paired.csv contém dados sobre duas medições: em Agosto e em Novembro em 13 lugares diferentes.

Use os seguintes testes e responda os itens abaixo:
- Test t pareado

- Wilcoxon signed rank.

1. Qual o mês com maior média e maior mediana?

2. A diferença é significativa?

3. Discuta a diferença dos p-valores. Quais dos 2 testes você acha mais apropriado nesse caso?



In [68]:
agosto = dados_pareado.August
novembro = dados_pareado.November

In [69]:
_, p1 = stats.ttest_ind(agosto, novembro)
print(p1)

if p1 > 0.05:
	print('Provavelmente é a mesma distribuição')
else:
	print('Estatisticamente diferentes')

0.03088448455801124
Estatisticamente diferentes


In [70]:
_, p2 = stats.ranksums(agosto, novembro)
print(p2)

if p2 > 0.05:
	print('Provavelmente é a mesma distribuição')
else:
	print('Estatisticamente diferentes')

0.01709698555220962
Estatisticamente diferentes


In [71]:
print("Agosto: {}".format(np.mean(agosto)))
print("Novembro: {}".format(np.mean(novembro)))

Agosto: 11.423076923076925
Novembro: 16.323076923076922


In [72]:
print("Agosto: {}".format(np.median(agosto)))
print("Novembro: {}".format(np.median(novembro)))

Agosto: 10.0
Novembro: 15.3


Ambos os testes acusaram que as amostras são estatisticamente diferentes, com p-value 0.03088448455801124 para t-student e p-value 0.01709698555220962 para wilcoxon. 

O mês de Novembro possui a média e mediana maior do que o mês de agosto, com o valor de ~16.32 para média e 15.3 para mediana.

O teste mais aconselhado é Wilcoxon. Como foram coletadas temperaturas dos mesmos locais em diferentes períodos de tempo, há uma relação de dados pareados, por conta da localização.

Mesmo que o teste T-Student tenha acusado diferença, o melhor seria Wilcoxon.

## Exercício 3

Use os dados (não pareados) do dataset bp.csv

Gere o intervalo de confiança para a pressão sangüínea dos pacientes com diabetes e sem diabetes (intervalo de 95% de confiança) usando o método de bootstrap

In [73]:
ic_diabeticos = stats.bootstrap((diabeticos.bp,), np.mean, confidence_level=0.95,
                                random_state=1, method='BCa')

ic_nao_diabeticos = stats.bootstrap((nao_diabeticos.bp,), np.mean, confidence_level=0.95, 
                                    random_state=1, method='BCa')

print("    IC diabéticos: {}".format(ic_diabeticos.confidence_interval))
print("IC não diabéticos: {}".format(ic_nao_diabeticos.confidence_interval))

    IC diabéticos: ConfidenceInterval(low=71.77941176470588, high=77.29411764705883)
IC não diabéticos: ConfidenceInterval(low=67.6590909090909, high=71.47727272727273)


Para os diabéticos, o intervalo de confiança é [71.77941176470588, 77.29411764705883].

Para os não diabéticos, o intervalo de confiança é [67.6590909090909, 71.47727272727273].

## Exercício 4

Teste Visual:  Há intersecção entre os intervalos de confiança dos 2 conjuntos de dados acima (com diabetes e sem diabetes)? Isso está de acordo com o teste de diferença significativa entre eles feito na questão 1?

Ambos os pacientes apresentaram média dentro do intervalo de confiança. Ou seja, com 95% de confiança posso concluir que o valor está dentro da média confiável. 

O valor da média para os diabéticos é 74.58823529411765 e o intervalo de confiança é [71.77941176470588, 77.29411764705883].

O valor da média para os diabéticos é 69.54545454545455 e o intervalo de confiança é [67.6590909090909, 71.47727272727273].

## Exercício 5

Calcule o Cohen D com o pooled standard deviation para a diferença entre a pressão sangüínea de pacientes com e sem diabetes.

In [74]:
from math import sqrt
 
def cohend(d1, d2):
	n1, n2 = len(d1), len(d2)
	s1, s2 = np.var(d1, ddof=1), np.var(d2, ddof=1)
	s = sqrt(((n1 - 1) * s1 + (n2 - 1) * s2) / (n1 + n2 - 2))
	u1, u2 = np.mean(d1), np.mean(d2)
	return (u1 - u2) / s

In [75]:
d = cohend(diabeticos.bp, nao_diabeticos.bp)

In [76]:
d

0.4480342648766951